frontend/src/components/common/StatusFilter.tsx

```diff
@@ -0,0 +1,138 @@
+import { Icon } from '@iconify/react';
+import Autocomplete from '@mui/material/Autocomplete';
+import Box from '@mui/material/Box';
+import Checkbox from '@mui/material/Checkbox';
+import { useTheme } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory, useLocation } from 'react-router-dom';
+import { addQuery } from '../../helpers';
+import { setStatusFilter } from '../../redux/filterSlice';
+import { useTypedSelector } from '../../redux/reducers/reducers';
+
+export interface PureStatusesAutocompleteProps {
+  statusNames: string[];
+  onChange: (event: React.ChangeEvent<{}>, newValue: string[]) => void;
+  filter: { statuses: Set<string>; search: string };
+}
+
+export function PureStatusesAutocomplete({
+  statusNames,
+  onChange: onChangeFromProps,
+  filter,
+}: PureStatusesAutocompleteProps) {
+  const theme = useTheme();
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [statusInput, setStatusInput] = React.useState<string>('');
+  const maxStatusesChars = 12;
+
+  const onInputChange = (event: object, value: string, reason: string) => {
+    // For some reason, the AutoComplete component resets the text after a short
+    // delay, so we need to avoid that or the user won't be able to edit/use what they type.
+    if (reason !== 'reset') {
+      setStatusInput(value);
+    }
+  };
+
+  const onChange = (event: React.ChangeEvent<{}>, newValue: string[]) => {
+    // Now we reset the input so it won't show next to the selected namespaces.
+    setStatusInput('');
+    onChangeFromProps(event, newValue);
+  };
+
+  return (
+    <Autocomplete
+      multiple
+      id="statuses-filter"
+      autoComplete
+      options={statusNames}
+      onChange={onChange}
+      onInputChange={onInputChange}
+      inputValue={statusInput}
+      // We reverse the namespaces so the last chosen appear as the first in the label. This
+      // is useful since the label is ellipsized and this we get to see it change.
+      value={[...filter.statuses.values()].reverse()}
+      renderOption={(props, option, { selected }) => (
+        <li {...props}>
```

Comment:
If at all possible, we avoid html tags and use Mui components. Could you please switch these ones out for mui components?